### PR TITLE
MAINT: Add test runners

### PR DESCRIPTION
**Pull request by bashtage**
Add test runners to all __init__ that have test directories below
reformat affected __init__ modules

xref #4245

- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html#writing-the-commit-message).

### CODE REVIEW

File: statsmodels/__init__.py [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-import sys
+from ._test_runner import PytestTester
+from .tools.sm_exceptions import (ConvergenceWarning, CacheWriteWarning,
+                                  IterationLimitWarning, InvalidTestWarning)
+from ._version import get_versions
 
-from warnings import simplefilter
-from statsmodels.tools.sm_exceptions import (ConvergenceWarning, CacheWriteWarning,
-                                             IterationLimitWarning, InvalidTestWarning)
+__docformat__ = 'restructuredtext'
```
Comment (jbrockmendel) [inline statsmodels/__init__.py]: Not for this PR, but are these lines still necessary?  They pop up in a bunch of places.

File: statsmodels/_test_runner.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import os
+import sys
+from distutils.version import LooseVersion
+
+
+class PytestTester(object):
```
Comment (jbrockmendel) [inline statsmodels/_test_runner.py]: We already have both tools.testing and tools._testing.  Can't this go in one of those?

File: statsmodels/formula/__init__.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 __all__ = ["handle_formula_data"]
```
Comment (jbrockmendel) [inline statsmodels/formula/__init__.py]: "test"?

File: statsmodels/_test_runner.py (hunk 2 above)
Comment (bashtage) [reply statsmodels/_test_runner.py]: Some care is needed to avoid circular imports, so maybe

File: statsmodels/__init__.py (hunk 1 above)
Comment (bashtage) [reply statsmodels/__init__.py]: Absolutely no idea.  I suspect not but don't need to fix everything now.

**Pull request merged**